### PR TITLE
[enh] Assert dpkg is not broken before app install

### PR DESCRIPTION
**Pull request by alexAubin**
## The problem

Follow-up of https://github.com/YunoHost/issues/issues/1284 and https://github.com/YunoHost/yunohost/pull/638.

## Solution

Make sure that dpkg/apt ain't broken before starting to install an app. This could also be done for other scripts which are quite likely to rely on apt : upgrade, remove and restore. I don't know if that's relevant though ... e.g. maybe we want people (or yunohost) to trigger removals even if dpkg/apt is broken ? I dunno ...

Note that the plan is anyway to add this check in ynh_apt though https://github.com/YunoHost/yunohost/pull/638 but the plan is to, where relevant, avoid starting to run a whole script if we can know it's gonna fail anyway.

## PR Status

Tested and working

## How to test

c.f. https://github.com/YunoHost/issues/issues/1284#issuecomment-459375469

## Validation

- [ ] Principle agreement 0/2 : 
- [ ] Quick review 0/1 : 
- [ ] Simple test 0/1 : 
- [ ] Deep review 0/1 :

### CODE REVIEW

**Comment by rds13** [comment]
I find it relevant that Yunohost doesn't try to install some packages on a broken system.

**Comment by alexAubin** [comment]
So today there was an interesting use case where : 
- someone installed the collaboraofficedocker app
- when uninstall it, it broke apt for some reason related to docker-ce and whatever
- later tried to upgrade nextcloud
- but apt was broken, so the upgrade failed
- and the restore of nextcloud's backup also failed, resulting in the app getting removed

TL;DR : one should not be able to try to upgrade an app if dpkg is broken or that will result in absolute disaster
